developer-docs-site/docs/guides/move-guides/mint-nft-cli.md

```diff
@@ -3,76 +3,188 @@ title: "Mint an NFT with Aptos CLI"
 
-1. Create an account on Aptos testnet by running the following command and selecting `testnet`:
 ```shell
-aptos init --profile nft-receiver
+    fun init_module(resource_account: &signer) {
+        let resource_signer_cap = resource_account::retrieve_resource_account_cap(resource_account, @source_addr);
+        let resource_signer = account::create_signer_with_capability(&resource_signer_cap);
 ```
 
-1. When prompted, select `testnet` by entering it:
+The `mint_nft` function of [`minting.move`](https://github.com/aptos-labs/nft-tutorial/blob/main/sources/minting.move) gets a collection and creates a token.
+
+With the resulting TokenData ID, the function uses the resource signer of the module to mint the token to the `nft-receiver`.
 
+For example:
 ```shell
-Configuring for profile nft-receiver
-Choose network from [devnet, testnet, mainnet, local, custom | defaults to devnet]
-testnet
+    public entry fun mint_nft(receiver: &signer) acquires ModuleData {
+        let receiver_addr = signer::address_of(receiver);
 ```
 
-1. When prompted for your private key, hit enter to generate a new key:
+### Signing
+
+The only argument taken by `mint_nft` is `receiver`. Any `entry fun` will take as the first parameter the type `&signer`. In both Move and Aptos, whenever you submit a transaction, the private key you sign the transaction with automatically makes the associated account the first parameter of the signer.
+
+You can go from the signer to an address but normally not the reverse. So when claiming an NFT, both the private keys of the minter and receiver are needed, as shown in the instructions below.
+
+In the `init_module`, the signer is always the account uploading the contract. This gets combined with:
+
 ```shell
-Enter your private key as a hex literal (0x...) [Current: None | No input: Generate new key (or keep one if present)]
+        token::create_collection(&resource_signer, collection, description, collection_uri, maximum_supply, mutate_setting);
+
 ```
+Where the `&resource_signer` is the first parameter, defined previously as a new address that has all of the attributes of the original account plus signer capability. See:
 
-1. Receive output resembling:
 ```shell
-No key given, generating key...
-Account blah does not exist, you will need to create and fun the account through a community faucet e.g. https://aptoslabs.com/testnet-faucet, or by transferring funds from another account
+        signer_cap: account::SignerCapability,
+```
 
----
-Aptos CLI is now set up for account blah as profile nft-receiver!  Run `aptos --help` for more information about commands
-{
-  "Result": "Success"
-}
-➜  devel
+The signer capability prevents anyone from getting the private key from the resource account. The [resource account](../resource-accounts.md) is entirely controlled by the contract.
```

Comment:
I think a resource account prevents anyone from getting the private key! the signer capability allows the module to sign autonomously

Reply:
Fixed! Thank you.